### ISSUE TITLE
Fix false positives for interpolation in font-family-no-duplicate-names

### ISSUE THREAD

**Issue by Andersos** [stylelint/stylelint#5200]
> Clearly describe the bug

There seem to be a bug for the rules `font-family-no-duplicate-names` and  `font-family-no-missing-generic-family-keyword` when using a variable to set font-family when writing css-in-js

> Which rule, if any, is the bug related to?
`font-family-no-duplicate-names`
`font-family-no-missing-generic-family-keyword`

> What code is needed to reproduce the bug?

```js
styled.li<{ primary?: "true" }>`
  font-family: ${(props) =>
    props.primary
      ? `"Company Font", "Helvetica Neue", Helvetica, Arial, sans-serif`
      : `"Helvetica Neue", Helvetica, Arial, sans-serif`};
`;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "font-family-no-duplicate-names": true,
    "font-family-no-missing-generic-family-keyword": true
  }
}
```

> Which version of stylelint are you using?

`13.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Both the VS Code Editor and CLI is warning of this error "lint:css": "stylelint './src/**/*.tsx'",

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Its related to css-in-js

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
    5:46 errorUnexpected missing generic font family (font-family-no-missing-generic-family-keyword)
    5:10 errorUnexpected duplicate name Helvetica Neue (font-family-no-duplicate-names)
    5:28 errorUnexpected duplicate name Helvetica (font-family-no-duplicate-names)
    5:39 errorUnexpected duplicate name Arial (font-family-no-duplicate-names)

```

It should be reproducable here also: 

https://stylelint.io/demo#N4Igxg9gJgpiBcIDOAXAngGxlAdBglgDzAAEADgE74C2AhhWgPzwkA6IKFArjOyQL4A+AAasAdiRIAzCGJQBaKbWr4MaFgBJgACkoQySAJQkAvIPGTJegzko16aC5cmMSokAAkYGAG4wU+GC0JAByMDzsADQkXr7+gbTRAIJUtBjRSLRiSPJIMFRSohLOJCzusX4BQaHhvCDRFfFByanpJJnZufn4hfwA3OLCffXIaHK0AB4I4Eg5+GLyAFZII5BiUvgA5tPATuzcWEjsLLvFkuwycorKqmjyYhDyUFxkBEEoMPfKMEcIJJw8SJOc4gS4KJQqNT3R4qWbzTbyTYwMTdMDXSF3ADWMDQAHcIBQoMd-txeMV+OJ+CB+EA

**Comment by jeddy3** [comment]
@Andersos Thanks for the report, using the template and reproducible demo.

It's likely a problem with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a list of the problems with the syntax that we need help to fix.

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time.

The issue may be resolved by [this experimental work into substitution-for-interpolation](https://github.com/stylelint/postcss-css-in-js/pull/46).

**Comment by Andersos** [comment]
Thanks for all the helpful links. Im happy the issue I am seeing was reproducable by others. I would love to help contribute but I might have to see if I can find a simpler task.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
